src/tool/core/helpers/problem_code.rs

```diff
@@ -136,6 +136,15 @@ impl FunctionInfo {
         names.join(", ")
     }
 
+    pub fn get_solution_comparison_code(&self) -> String {
+        if let Some(FunctionArgType::F64) = &self.return_type {
+            "assert!((actual - expected).abs() < 1e-5, \"Assertion failed: actual {actual:.5} but expected {expected:.5}. Diff is more than 1e-5.\");"
+        } else {
```

Comment:
```rust
else if  Some(FunctionArgType::VecF64) = &self.return_type {
            r#"
            assert!(actual.len(), expected.len(), "actual {actual} not same length as {expected}");
            actual.iter().zip(expected.iter()).enumerate().for_each(|(i, (actual, expected))| {
            assert!((actual - expected).abs() < 1e-5, "Assertion failed: {i} the element in actual {actual:.5} but expected {expected:.5}. Diff is more than 1e-5.");
           })
            "#
} else
```

Reply:
may be match statement is better ?

Reply:
That's what I had and then got recommended to use an if let by clippy. When we add more we'll change it to a match for now I think this is fine

Reply:
> ```rust
> else if  Some(FunctionArgType::VecF64) = &self.return_type {
>             r#"
>             assert!(actual.len(), expected.len(), "actual {actual} not same length as {expected}");
>             actual.iter().zip(expected.iter()).enumerate().for_each(|(i, (actual, expected))| {
>             assert!((actual - expected).abs() < 1e-5, "Assertion failed: {i}th element in actual {actual:.5} but expected {expected:.5}. Diff is more than 1e-5.");
>            })
>             "#
> } else
> ```

Oh you are doing both. Yeah better to change it to a match. Is there a question where that is the return type?

Reply:
created #33! resolving